utils.go

```diff
@@ -49,10 +49,12 @@ func promptForYes(message string) bool {
 	return strings.HasPrefix(yn, "Y") || strings.HasPrefix(yn, "y")
 }
 
-func generate_secret(length int) (string, error) {
```

Comment:
let's create a unit test for this.
also because I suspect this implementation won't work when secretChars contains wide unicode characters ( btw in Go there are no chars, they are called runes).

secretChars: "☑☠♨⛱"

Reply:
Yes, this indeed did only work for single byte characters. I'm adding some table tests, feel free to suggest more.